rotkehlchen/globaldb/handler.py

```diff
@@ -349,6 +378,28 @@ def _prepare_search_assets_query(filter_query: 'AssetsFilterQuery') -> Tuple[str
         query = parent_query + query
         return query, bindings
 
+    @staticmethod
+    def _handle_eth2_as_eth_setting_in_assets_search(
+            entry: Tuple[str, str, str],
+            search_result: List,
+            treat_eth2_as_eth: bool,
+            found_eth: bool,
+            levenshtein_dist: Optional[int] = None,
+            levenshtein_distances_list: Optional[List] = None,
+    ) -> bool:
+        """Utility function that ensures search results respect the `treat_eth2_as_eth` setting."""
+        if treat_eth2_as_eth is True and entry[0] in (A_ETH.identifier, A_ETH2.identifier):  # noqa:E501
```

Comment:
okay so I see why you tried to create this helper function but this is an overkill. It makes things much harder to read and does not save space since you use it only 2 times and pass different optional arguments in each location so you add extra if condition checks in each place.

Please just incorporate this in the actual places where you call this.

Also ... I think there is a bug here. This should return immediately `if found_eth is True`, no ?

Generally I think you can make it much simpler and easier to read if you add a separate check in the 2 functions.